kahuna/public/js/services/preset-label.js

```diff
@@ -0,0 +1,39 @@
+import angular from 'angular';
+
+var presetLabelService = angular.module('kahuna.services.presetLabel', []);
+
+presetLabelService.factory('presetLabelService',
+                            ['$window', function ($window) {
+
+    const presetLabelsKey = 'preset labels';
+
+    function getLabels() {
+        return JSON.parse($window.localStorage.getItem(presetLabelsKey));
+    }
+
+    function addLabels(newLabels) {
+        let labels = new Set(getLabels());
+        newLabels.map( newLabel => labels.add(newLabel) );
```

Comment:
It's generally best practice to use `forEach` when we aren't expecting the result of a `map`